### PR TITLE
feat: add support for PSC connections

### PR DESCRIPTION
**Pull request by jackwotherspoon**
Add a new dialOption for `WithPSC()` to support connections to Cloud SQL using [private service connect](https://cloud.google.com/vpc/docs/private-service-connect).

### CODE REVIEW

**Comment by jackwotherspoon** [comment]
This currently gets the error: 
```
dial tcp: lookup xxxxxxxxxxxx.xxxxxxxxxxxx.us-central1.sql.goog. on 127.0.0.1:53: no such host)
```

Same DNS name is reachable with Cloud SQL Python Connector so definitely is something needed adjustment in Go connector.

File: internal/cloudsql/refresh_test.go [hunk 1]
```diff
@@ -81,6 +83,13 @@ func TestRefresh(t *testing.T) {
 	if wantPrivateIP != gotIP {
 		t.Fatalf("metadata IP mismatch, want = %v, got = %v", wantPrivateIP, gotIP)
 	}
+	gotPSC, ok := rr.ipAddrs[PSC]
+	if !ok {
+		t.Fatalf("metadata IP addresses did not include PSC endpoint")
```
Comment (enocom) [inline internal/cloudsql/refresh_test.go:88]: t.Fatal

File: internal/mock/cloudsql.go [hunk 2]
```diff
@@ -155,6 +163,7 @@ func NewFakeCSQLInstance(project, region, name string, opts ...FakeCSQLInstanceO
 		region:       region,
 		name:         name,
 		ipAddrs:      map[string]string{"PUBLIC": "0.0.0.0"},
+		DNSName:      "",
```
Comment (enocom) [inline internal/mock/cloudsql.go:166]: Should we add a good default here?

**Review by enocom — APPROVED** [review]
Some minor nits. LGTM.

File: internal/cloudsql/refresh_test.go (hunk 1 above)
Comment (jackwotherspoon) [reply internal/cloudsql/refresh_test.go]: Done

File: internal/mock/cloudsql.go (hunk 2 above)
Comment (enocom) [reply internal/mock/cloudsql.go:166]: Sounds good to me.

**Pull request merged**